moto/mediaconnect/models.py

```diff
@@ -0,0 +1,191 @@
+        self.source = kwargs.get("source", {})
+        self.source_failover_config = kwargs.get("source_failover_config", {})
+        self.sources = kwargs.get("sources", [])
+        self.vpc_interfaces = kwargs.get("vpc_interfaces", [])
+        self.status = "STANDBY"  # one of 'STANDBY'|'ACTIVE'|'UPDATING'|'DELETING'|'STARTING'|'STOPPING'|'ERROR'
+        self._previous_status = None
+        self.description = None
+        self.flow_arn = None
+        self.egress_ip = None
+
+    def to_dict(self, include=None):
+        data = {
+            "availabilityZone": self.availability_zone,
+            "description": self.description,
+            "egressIp": self.egress_ip,
+            "entitlements": self.entitlements,
+            "flowArn": self.flow_arn,
+            "name": self.name,
+            "outputs": self.outputs,
+            "source": self.source,
+            "sourceFailoverConfig": self.source_failover_config,
+            "sources": self.sources,
+            "status": self.status,
+            "vpcInterfaces": self.vpc_interfaces,
+        }
+        if include:
+            new_data = {k: v for k, v in data.items() if k in include}
+            if "sourceType" in include:
+                new_data["sourceType"] = "OWNED"
+            return new_data
+        return data
+
+    def resolve_transient_states(self):
+        if self.status in ["STARTING"]:
+            self.status = "ACTIVE"
+        if self.status in ["STOPPING"]:
+            self.status = "STANDBY"
+        if self.status in ["UPDATING"]:
+            self.status = self._previous_status
+            self._previous_status = None
+
+
+class Resource(BaseModel):
+    def __init__(self, *args, **kwargs):
+        self.resource_arn = kwargs.get("resource_arn")
+        self.tags = OrderedDict()
+
+    def to_dict(self):
+        data = {
+            "resourceArn": self.resource_arn,
+            "tags": self.tags,
+        }
+        return data
+
+
+class MediaConnectBackend(BaseBackend):
+    def __init__(self, region_name=None):
+        super(MediaConnectBackend, self).__init__()
+        self.region_name = region_name
+        self._flows = OrderedDict()
+        self._resources = OrderedDict()
+
+    def reset(self):
+        region_name = self.region_name
+        self.__dict__ = {}
+        self.__init__(region_name)
+
+    def create_flow(
+        self,
+        availability_zone,
+        entitlements,
+        name,
+        outputs,
+        source,
+        source_failover_config,
+        sources,
+        vpc_interfaces,
+    ):
+        flow = Flow(
+            availability_zone=availability_zone,
+            entitlements=entitlements,
+            name=name,
+            outputs=outputs,
+            source=source,
+            source_failover_config=source_failover_config,
+            sources=sources,
+            vpc_interfaces=vpc_interfaces,
+        )
+        flow.description = "A Moto test flow"
```

Comment:
Just curious - what's the value that AWS outputs here?
Strange that there's no way to pass this during creation

Reply:
I wondered about this too. Seems to be inconsistent. AWS docs say: "This description appears only on the MediaConnect console and is not visible outside of the current AWS account. "

So I'm assuming the API doesn't support setting the description in the request, but you can set it manually from the AWS console, and if you do, you'll see it in the Describe response. 

On the other hand, for Entitlements, Sources and Outputs of a flow you _can_ provide descriptions in the request.